test/base/testing_brave_browser_process.cc

```diff
@@ -53,7 +58,17 @@ TestingBraveBrowserProcess::~TestingBraveBrowserProcess() = default;
 void TestingBraveBrowserProcess::StartBraveServices() {}
 
 brave_shields::AdBlockService* TestingBraveBrowserProcess::ad_block_service() {
-  DCHECK(ad_block_service_);
+  if (!ad_block_service_) {
+    scoped_refptr<base::SequencedTaskRunner> task_runner(
+        base::ThreadPool::CreateSequencedTaskRunner(
+            {base::MayBlock(), base::TaskPriority::USER_BLOCKING,
+             base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN}));
+    ad_block_service_ = std::make_unique<brave_shields::AdBlockService>(
```

Comment:
@antonok-edm it would be nice if we had AdBlockService/AdBlockServiceImpl so we can provide a fake AdBlockService here